### PR TITLE
[CouchDB] $port should be an int not a string

### PR DESCRIPTION
**Pull request by maltheism**
### Brief summary of changes

The $port should be an int but $couchConfig returns a string. I added the intval function when using the $couchConfig function to retrieve the port from config.xml. I also updated the comment for the constructor to reflect an int for $port and not a string.

### CODE REVIEW

File: php/libraries/CouchDB.class.inc [hunk 1]
```diff
@@ -199,14 +199,14 @@ class CouchDB
      *
      * @param string $database The couchDB database name
      * @param string $host     The couchDB host name
-     * @param string $port     The couchDB port number
+     * @param int    $port     The couchDB port number
```
Comment (johnsaigle) [inline php/libraries/CouchDB.class.inc]:
```suggestion
     * @param ?int    $port     The couchDB port number
```

File: php/libraries/CouchDB.class.inc [hunk 2]
```diff
@@ -199,14 +199,14 @@ class CouchDB
...
      * @param string $user     The couchDB user name
      * @param string $password The couchDB password
      */
     private function __construct(
         string $database = '',
         string $host = '',
-        string $port = '',
+        int $port = null,
```
Comment (johnsaigle) [inline php/libraries/CouchDB.class.inc]:
```suggestion
        ?int $port = null,
```

File: php/libraries/CouchDB.class.inc [hunk 3]
```diff
@@ -199,14 +199,14 @@ class CouchDB
...
-     * @param string $port     The couchDB port number
+     * @param ?int   $port     The couchDB port number
      * @param string $user     The couchDB user name
      * @param string $password The couchDB password
      */
     private function __construct(
         string $database = '',
         string $host = '',
-        string $port = '',
+        ?int   $port = null,
```
Comment (driusan) [inline php/libraries/CouchDB.class.inc]: What does it mean to have a null port? I think this should be non-nullable and have a default of 5984 (CouchDB's default port)

**Comment by ridz1208** [comment]
@driusan @maltheism this should have gone to bugfix

**Comment by driusan** [comment]
@ridz1208 bugfix doesn't have any type hinting, does it?

**Comment by ridz1208** [comment]
@driusan it does for this
thats why we are having these issues on releases like 20.1.2 and 20.2.0.

Not sure however why CCNA is not having that problem but others are (not sure if there is a global PHP setting for Types-strict or not...

**Comment by driusan** [comment]
@ridz1208 okay, then it should have went to bugfix.. it should be a simple fix to backport.

**Pull request merged**